CMakeLists.txt

```diff
@@ -54,24 +53,30 @@ add_library(graphqlservice
   GraphQLResponse.cpp
   GraphQLService.cpp
   Introspection.cpp
-  ${CMAKE_BINARY_DIR}/graphqlservice/IntrospectionSchema.cpp)
+  IntrospectionSchema.cpp)
 target_link_libraries(graphqlservice PRIVATE taocpp::pegtl)
 target_link_libraries(graphqlservice PUBLIC Threads::Threads)
-target_include_directories(graphqlservice
-  PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>
-    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
-    $<INSTALL_INTERFACE:include>
-)
+# Make system includes (e.g. <graphqlservice/public.h>) work relative to the build/install generator expressions
+target_include_directories(graphqlservice SYSTEM PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>
+  $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/include>
+  $<INSTALL_INTERFACE:include>)
+# Make local includes (e.g. "private.h") work while building graphqlservice, but don't export them
+target_include_directories(graphqlservice PRIVATE
+  ${CMAKE_SOURCE_DIR}/include
+  ${CMAKE_BINARY_DIR}/include)
 cppgraphqlgen_target_set_cxx_standard(graphqlservice)
 
 option(BUILD_TESTS "Build the tests and sample schema library." ON)
 option(UPDATE_SAMPLES "Regenerate the sample schema sources whether or not we're building the tests and the sample library." ON)
 
 if(BUILD_TESTS OR UPDATE_SAMPLES)
   add_custom_command(
-    OUTPUT TodaySchema.cpp TodaySchema.h
+    OUTPUT
+      TodaySchema.cpp
+      include/TodaySchema.h
     COMMAND schemagen ${CMAKE_SOURCE_DIR}/schema.today.graphql Today today
+    COMMAND ${CMAKE_COMMAND} -E rename TodaySchema.h include/TodaySchema.h
```

Comment:
Note: by the way, this would be a useful option to schemagen, to specify the output directory somehow.